src/js/components/pages/adjustable-grid.jsx

```diff
@@ -117,8 +112,20 @@ var RespGrid = React.createClass({
     restore: function (index, item) {
         SceneActions.restoreComp(index, item);
     },
+
+    // APEP Calculate the height each time this is called
     max: function (index, item) {
-        var maxHeightValue = this.state.parentHeight ? this.state.parentHeight / 30 : 30;
+        // APEP calculate the height of the screen using the parent component.  This is fixed to the max size of screen
+        // Library uses this maths for calculating height of a component heightPx * h + (marginH * (h - 1))
+        var margin = 10;
+        var dom = ReactDOM.findDOMNode(this);
+        var rowHeight = dom.parentElement.clientHeight / this.state.rows;
+        var elementHeightPlusMargin = dom.parentElement.clientHeight + margin;
+        var rowHeightPlusMargin = rowHeight + margin;
+
+        // Rearranged h = (elementHeight + marginH) / (heightPx + marginH)
+        var maxHeightValue = Math.floor(elementHeightPlusMargin/rowHeightPlusMargin);
```

Comment:
As height is required to be an integer, we should always round down to ensure the h is not above the full screen.